Detectors/GlobalTracking/src/MatchHMP.cxx

```diff
@@ -377,41 +398,39 @@ void MatchHMP::doMatching()
       auto& trefTrk = trackWork.first;
 
       prop->getFieldXYZ(trefTrk.getXYZGlo(), bxyz);
-      Double_t bz = -bxyz[2];
+      double bz = -bxyz[2];
 
-      double timeUncert = 100.; //= trackWork.second.getTimeStampError();
+      double timeUncert = trackWork.second.getTimeStampError();
 
-      float minTrkTime = (trackWork.second.getTimeStamp() - mSigmaTimeCut * timeUncert) * 1.E3; // minimum track time in ns
-      float maxTrkTime = (trackWork.second.getTimeStamp() + mSigmaTimeCut * timeUncert) * 1.E3; // maximum track time in ns
+      float minTrkTime = (trackWork.second.getTimeStamp() - mSigmaTimeCut * timeUncert);
+      float maxTrkTime = (trackWork.second.getTimeStamp() + mSigmaTimeCut * timeUncert);
 
-      if (evtTime < (maxTrkTime + timeFromTF) && evtTime > (minTrkTime + timeFromTF)) {
+      // if (evtTime < (maxTrkTime + timeFromTF) && evtTime > (minTrkTime + timeFromTF)) {
+      if (evtTime < maxTrkTime && evtTime > minTrkTime) {
 
         evtTracks++;
 
-        MatchInfo* matching = new o2::dataformats::MatchInfoHMP(999999, mTrackGid[type][cacheTrk[itrk]]);
+        MatchInfo matching = MatchInfo(999999, mTrackGid[type][cacheTrk[itrk]]);
 
-        matching->setHMPIDtrk(0, 0, 0, 0);            // no intersection found
-        matching->setHMPIDmip(0, 0, 0, 0);            // store mip info in any case
-        matching->setIdxHMPClus(99, 99999);           // chamber not found, mip not yet considered
-        matching->setHMPsignal(Recon::kNotPerformed); // ring reconstruction not yet performed
-        matching->setIdxTrack(trackGid);
+        matching.setHMPIDtrk(0, 0, 0, 0);            // no intersection found
+        matching.setHMPIDmip(0, 0, 0, 0);            // store mip info in any case
+        matching.setIdxHMPClus(99, 99999);           // chamber not found, mip not yet considered
+        matching.setHMPsignal(Recon::kNotPerformed); // ring reconstruction not yet performed
+        matching.setIdxTrack(trackGid);
 
-        TrackHMP* hmpTrk = new TrackHMP(trefTrk); // create a hmpid track to be used for propagation and matching
-        TrackHMP* hmpTrkConstrained = nullptr;    // create a hmpid track to be used for propagation and matching
+        TrackHMP hmpTrk = TrackHMP(trefTrk); // create a hmpid track to be used for propagation and matching
```

Comment:
use simply ` TrackHMP hmpTrk(trefTrk);` to avoid extra assignment.

Reply:
Sure, of course it is wrong! By mistake the push it is outside the if() ...! IThanks to have spotted it!